Vertically align mail icon with text

KentShikama opened the pull request:
It now looks as follows:

![screen shot 2015-06-01 at 11 48 13 pm](https://cloud.githubusercontent.com/assets/6329898/7916186/2fc919be-08bd-11e5-99a5-9d6fd47fe7c1.png)

Commit:
Vertically align mail icon with text

jhass commented:
Uh, that affects none of the other icons?

AugierLe42e commented:
Meh... I think it does. Completely unverified, but I think it will mess with the trash icon at least.

theworldbright commented:
Let me check. I just realized that that scss styling isn't as isolated as I thought.

AugierLe42e commented:
Haha, nope, it isn't. Actually, here, you have modified the styling of the entypo icons for the whole application :p
If I'm not wrong, SCSS in `new_styles` directory define rules for the whole application like button styling.

Commit:
Remove unnecessary explicit div tag

Commit:
Vertically align entypo mail specifically

This commit specifically aligns entypo mail
unlike before. There were only two instances that
were affected that I checked: the forgot password
page and the sign up page. The commit also does
some slight adjustments to the sign up and
login form by raising the nth entypo icon by a
slight amount.

Commit:
Minor refactoring

theworldbright commented:
Here is what the forgot your password page looks like:

![screen shot 2015-06-02 at 1 28 23 am](https://cloud.githubusercontent.com/assets/6329898/7917636/c3b277a2-08c6-11e5-8aa6-ae3bc55b0a62.png)

Here is what the login page looks like:

![screen shot 2015-06-02 at 1 30 12 am](https://cloud.githubusercontent.com/assets/6329898/7917684/02fcae6e-08c7-11e5-9923-d9083f0512a7.png)

Here is what the signup page looks like:

![screen shot 2015-06-02 at 1 32 01 am](https://cloud.githubusercontent.com/assets/6329898/7917720/3dc692ee-08c7-11e5-8d98-117df40edfde.png)

theworldbright commented:
I searched all occurrences of '.entypo.mail' in the project. There were three results, and the signup page and the forgot your password page were the only two places where the entypo mail icon was used in a block-form > fieldset.

jhass commented:
I guess we can live with that, but if you want it perfect I think either the vertical or the bottom baseline should match.... :P

Current: ![](http://cloud.aeshna.de/u/mrzyx/screenshots/screenshot_20150601_184029.png)
Your screenshot: ![](http://cloud.aeshna.de/u/mrzyx/screenshots/screenshot_20150601_184214.png)

AugierLe42e commented on app/assets/stylesheets/new_styles/_forms.scss:
```diff
@@ -86,18 +86,22 @@ form.block-form {
+      top: 87px;
     }
 
     .entypo:nth-of-type(4) {
```
Oh crap, that's so an ugly selector :/
What does it corresponds to ?

Flaburgan replied:
The fourth item. Icons are placed from the top so it's the cleanest solution I found.

AugierLe42e replied:
K, I trust you :p
As I said, I suck at JS :D

theworldbright commented:
I find it weird that the username icon is out of place. I didn't make any changes in my code that would affect that icon. It seems like in the develop branch both icons are aligned;  however, in the latest bootstrap3 branch (without my changes in these PR), the icons are slightly lower than they should be. Furthermore, I did a diff across the two branches and there has been no changes to the _form.scss. I will  continue to look but if anyone can give me a tip as to why this is happening, I would be interested to know.

AugierLe42e commented:
@theworldbright : It is likely possible that BS3 just broke everything. E.g: By default, all div have a null height, on the contrary to BS2, which introduced weird sides effects I had hard time to deal with when working on this (headache, bitch ! :p)

theworldbright commented:
@AugierLe42e Thanks for the tip. Yup it seems like there was a change in line height from BS2 to BS3. BS2 uses a line height of 20px while BS3 uses a line height of 1.42857143, causing a shift is all icon heights. For now, I'll just change the line height back to 20px.

AugierLe42e commented:
> For now, I'll just change the line height back to 20px.

Don't. [A PR has just been merged](https://github.com/diaspora/diaspora/commit/781ca2b76e7c6083978c9cf4a5f47274cb0c9e48) making font size a global Bootstrap variable. I wonder it it could solve your problem. Try to update your branch and see if it changes something.

theworldbright commented:
Yeah just saw that PR myself.

theworldbright commented:
I am now tempted to change the following line in bootstrap-variables.scss

```
// $line-height-base:        1.428571429 // 20/14
```

to 20px. Do you know if that would have any major side-effects off the top of your head?

jhass commented:
No expert either here, but yeah, I'd expect it to affect about everything.

AugierLe42e commented:
That could be a good idea. 1.428571429 seems to be 20/14, that is `baseline height / font size`. As our font size is 13px and not 14, that would be compilant

theworldbright commented:
Did some reading and it seems like using the "unitless" line height is recommended as it computes line heights based on the font size. It seems like overriding the value locally to 20px does the job, so I'll leave it at that for now.

@AugierLe42e Are you suggesting to change the value to 20/13 ~= 1.5...?

AugierLe42e commented:
> Are you suggesting to change the value to 20/13 ~= 1.5...?

Yep, I think the unit of this line height variable is in `em` or `rem` that is, a unit which is proportionnal to the body height of the font. So, it is a good practice to choose a size that is related to our font size (13px) than Bootstrap's default (14px).
